artifacts.go

```diff
@@ -121,7 +121,8 @@ func (artifact *S3Artifact) CreateTempFileForPUTBody() string {
 	baseName := filepath.Base(rawContentFile)
 	tmpFile, err := ioutil.TempFile("", baseName)
 	if err != nil {
-		log.Fatal(err)
+		incidentID := monitor.ReportError(err, "Could not create temporary file")
+		log.Fatalf("%v, incident ID: %s", err, incidentID)
```

Comment:
I've only just realised this myself, but I have a few places in the code where I am calling e.g. `log.Fatal` which should really just panic, so that the worker shutdown etc is handled gracefully (since we catch the panics and handle them). This PR is about something else, so I won't ask you to fix all my broken code, but I'll do that myself once this lands. In other words, where there are currently `log.Fatal[f]` function calls that execute _in the main go routine_, I'll change them to `panic` statements instead.